### PR TITLE
auto-docs for github pages

### PR DESCRIPTION
**Pull request by binaryaaron**
I've set up a build process for generating API documentation for the repo. 

The branch `gh-pages` will hold nothing but the HTML and related files needed to build the pages, and to ease the process of building / deploying those pages, there is a small script that automates the worst part of this process in `doc_builder.sh`. 

Google-style docstring examples are present in `tweet_checking.py` as well as a few small changes to the code structure and layout. I apologize for bundling changes into one big PR, but it feels necessary to do that all at this time.  

@fionapigott - it's probably best to merge this PR back into master before continuing your docstring updates.

I also would like specific feedback regarding local building / testing of documentation. 

Thanks!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -58,6 +58,36 @@ Tweet format changes. This option is enabled with `--do_format_checking` on the
...
+pip install sphinx
+pip install sphinx_bootstrap_theme
+```
+
+### Build
+
+```
+cd tweet_parser/docs
+make clean
+make html
```
Comment (jrmontag) [inline README.md]: In a clean env, I got the error below; sounds like some part of sphinx. Any insight into it before I go off and google around?

```
(doc-test)[tw-172-17-72-168 docs (doc_builder)]$ make clean
Removing everything under 'build'...
(doc-test)[tw-172-17-72-168 docs (doc_builder)]$ make html
Running Sphinx v1.6.3
making output directory...
loading pickled environment... not yet created
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 5 source files that are out of date
updating environment: 5 added, 0 changed, 0 removed
reading sources... [ 20%] README
Extension error:
Could not import recommonmark.parser (needed for source parser) (exception: No module named 'recommonmark')
make: *** [html] Error 1
```

Comment (binaryaaron) [inline README.md]: That's on me -- I forgot to add recommonmark to the install instructions. I'll update and push back.

File: README.md [hunk 2]
```diff
@@ -58,6 +58,36 @@ Tweet format changes. This option is enabled with `--do_format_checking` on the
 command line, and by setting the keyword argument `do_format_checking` to
 `True` when initializing a `Tweet` object.
 
+## Documentation
```
Comment (jrmontag) [inline README.md]: Thanks for updating the installs in the documentation readme @binaryaaron. 

This also happen to highlight the fact that there are redundant instructions between the main repo readme and the docs readme. Does it make sense to completely separate the concerns? It doesn't seem necessary to have someone go through the doc build if they're submitting a bugfix in the code. If you think that's sensible, perhaps slim down the "Documentation" part of the main readme to just point to the longer one? 

I'm open to other ideas, too.

Comment (binaryaaron) [inline README.md]: I agree completely - let's separate the docs instructions from the overall repo instructions.

**Comment by binaryaaron** [comment]
@fionapigott - I updated the tests to reflect the naming changes in `tweet_checking`. Was there somewhere else where the changes were problematic?

**Comment by jrmontag** [comment]
@binaryaaron what should the protocol be for verifying the doc build? For example, I just ran the updated instructions, but got a couple of warnings (below). Do these represent things that need to be fixed, or just good-to-know-about warnings? 

```
(doc-test)[tw-172-17-72-168 docs (doc_builder)]$ make clean
Removing everything under 'build'...
(doc-test)[tw-172-17-72-168 docs (doc_builder)]$ make html
Running Sphinx v1.6.3
making output directory...
loading pickled environment... not yet created
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 5 source files that are out of date
updating environment: 5 added, 0 changed, 0 removed
reading sources... [100%] tweet_parser.getter_methods
/Users/jmontague/workspace/tweet_parser/docs/source/modules.rst:4: WARNING: toctree contains reference to nonexisting document 'setup'
/Users/jmontague/workspace/tweet_parser/tweet_parser/tweet.py:docstring of tweet_parser.tweet.Tweet.embedded_tweet:0: WARNING: Inline strong start-string without end-string.
WARNING: /Users/jmontague/workspace/tweet_parser/tweet_parser/tweet_checking.py:docstring of tweet_parser.tweet_checking.get_all_keys:19: (ERROR/3) Unknown directive type "todo".

.. todo:: check for ordering?
looking for now-outdated files... none found
pickling environment... done
checking consistency... /Users/jmontague/workspace/tweet_parser/docs/source/modules.rst: WARNING: document isn't included in any toctree
done
preparing documents... done
writing output... [100%] tweet_parser.getter_methods
generating indices... genindex py-modindex
writing additional pages... search
copying static files... done
copying extra files... done
dumping search index in English (code: en) ... done
dumping object inventory... done
build succeeded, 4 warnings.

Build finished. The HTML pages are in build/html.
```

Separately, with what frequency should we rebuild and push the gh-pages branch? Just when changes are made to the docstrings and examples, or something else?

**Comment by binaryaaron** [comment]
@jrmontag - 

I fixed those warnings. As far as I can tell, "testing" for the documentation is mostly inspection to ensure changes are reflected in the final HTML output. 

To address your other point:
I think there will be a bit of tinkering with the docs process. For right now, we'll stabilize on a 'current' set of documentation when we get @fionapigott's upcoming changes merged; after that it seems sane to only update docs in a few cases:

- functionality is added that will add a new docstring
- additions made to current functions

Which are linked to updating the API / repo itself. We should work with the greater org to discuss where this might fit into the greater ecosystem of dev.twitter.com and how many more examples we might add. I expect this to be a somewhat organic process.

**Comment by jrmontag** [comment]
Great, thanks for the fix and the context; all sounds good to me.

**Pull request merged**